tests/SAML2/CustomCondition.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace SAML2;
+
+use DOMElement;
+use SAML2\XML\saml\Condition;
+use Webmozart\Assert\Assert;
+
+final class CustomCondition extends Condition
```

Comment:
Right now I don't remember the exact approach we took before, but in this case, wouldn't it be enough to do this in `Condition`? Wouldn't the same approach of `BaseID` work here?

Reply:
I think we came to the conclusion that our BaseID implementation was wrong, because it doesn't consider any child-elements and/or XML-attributes... Not sure though

Reply:
Uhm, isn't it possible in those cases to just extend `BaseID` to implement whatever you need as child elements/attributes?

Reply:
I don't recall exactly.. Anyway, for BaseID we also created a CustomID test-case..